packages/hierarchies/src/hierarchies/internal/operators/Partition.ts

```diff
@@ -0,0 +1,52 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { from, Observable, ObservableInput, ReplaySubject } from "rxjs";
+
+/** @internal */
+export function partition<T, U extends T>(source: ObservableInput<T>, predicate: (value: T) => value is U): [Observable<U>, Observable<Exclude<T, U>>];
+/** @internal */
+export function partition<T>(source: ObservableInput<T>, predicate: (value: T) => boolean): [Observable<T>, Observable<T>];
+
+/**
+ * This is similar to `rxjs` `partition` operator, but it subscribes to source observable only once.
+ * @internal
+ */
+export function partition<T>(source: ObservableInput<T>, predicate: (value: T) => boolean): [Observable<T>, Observable<T>] {
```

Comment:
The whole point of introducing our own implementation was to avoid having to traverse the source observable twice. BTW, my implementation only subscribes to source when consumer subscribes to one of the returned observables.

Reply:
Oh ok, I thought the main problem was that RXJS subscribes 2 times to the same observable without sharing the replay